### PR TITLE
Test: Automation test phase 2

### PR DESCRIPTION
**Pull request by arun3528**

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 1]
```diff
@@ -183,11 +211,50 @@ export default class LocusInfo extends ChildEmitter {
...
+        this.emit(EVENTS.DESTROY_MEETING);
+      }
+      else if (this.fullState.removed) {
+        // user has been dropped from a meeting
         this.emit(EVENTS.DESTROY_MEETING);
       }
     }
     else {
-      // TODO: Check non 1:1 meeting
+      console.log('UNKNOWN MEETING TYPE');
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: we are revamping logging next

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 2]
```diff
@@ -360,13 +439,14 @@ export default class LocusInfo extends ChildEmitter {
       }
 
       // TODO: check if we need to save the sipUri here as well
-      this.updateMeeting(SelfUtils.getSipUrl(this.getLocusPartner(participants, self), this.parsedLocus.info));
+      // this.emit(LOCUSINFO.EVENTS.MEETING_UPDATE, SelfUtils.getSipUrl(this.getLocusPartner(participants, self), this.parsedLocus.fullState.type, this.parsedLocus.info.sipUri));
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: +1

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js [hunk 3]
```diff
@@ -20,21 +20,21 @@ InfoUtils.parsePolicy = (info) => {
...
-  if (info.locusTags && info.locusTags.indexOf(MEETING) >= 0) {
-    type = MEETING;
-  }
-  // For webex meeting claimed or unclaimed or calender meeting with webex site
-  if (info.webexServiceType === MEETING_CENTER || info.isPmr) {
-    return WEBEX_MEETING;
-  }
-  return type;
-};
+// InfoUtils.parseMeetingType = (info) => {
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js]: +1

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import btoa from 'btoa';
 import url from 'url';
+import btoa from 'btoa';
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js]: Okay, I'll work on that when i update constants.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 5]
```diff
@@ -203,11 +201,10 @@ export default class Meeting extends StatelessSparkPlugin {
      * @public
      * @memberof Meeting
      */
-    this.state = MeetingStateMachine.create(this);
+    this.mtgState = MeetingStateMachine.create(this);
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Hmmmmm. okay, i'll see if i can think of a better name. might call it meetingFSM (finite state machine, as it's mostly an internal tool for debugging.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 6]
```diff
@@ -47,7 +48,7 @@ export const IN_MEETING = 'IN_MEETING';
 export const INCOMING = 'INCOMING';
 export const INTENT_TO_JOIN = 2423005;
 
-
+export const _JOIN_ = 'JOIN';
 export const JOINED = 'JOINED';
 export const JOIN = 'join';
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/constants.js]: yeah ill revamp constants

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 7]
```diff
@@ -507,6 +524,8 @@ export const DEFAULT_TRANSFORM_REGEX = {
     }
   }
 };
+
+export const ROAP_SEQ_PRE = -1;
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/constants.js]: +1

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 8]
```diff
@@ -110,11 +110,37 @@ export default class LocusInfo extends ChildEmitter {
    * @returns {object} null
    */
   onFullLocus(locus, eventType) {
+    if (!locus) {
+      console.log('locus object undefined');
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: revamping logging next, not in this PR

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js [hunk 9]
```diff
@@ -110,11 +110,37 @@ export default class LocusInfo extends ChildEmitter {
...
+    }
     console.log('EVENTS', eventType);
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: revamping logging next

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js]: actually...i don't think i will -- the eslint rules don't really allow us to follow a typical alphabetically sorted import list...we can discuss in standup tomorrow.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 10]
```diff
@@ -67,7 +67,7 @@ export const ACTIVE = 'ACTIVE';
 
 export const BUSY = 'BUSY';
 
-export const CALL = 'CALL';
+export const _CALL_ = 'CALL';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: Is there something more descriptive we can call this? Or group it as a property within another object, like `LOCUS.CALL`?

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 11]
```diff
@@ -16,7 +16,23 @@ import Members from '../members/index';
...
+  BUSY,
+  INCOMING,
+  ONLINE,
+  _CALL_,
+  CONTENT,
+  ROAP_SEQ_PRE,
+  LOCUSINFO,
+  MEETING_STATE,
+  MEETING_STATE_MACHINE,
+  _JOIN_
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Same thing here.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 12]
```diff
@@ -72,7 +72,7 @@ export const CREATED = 'CREATED';
 export const CONFLICT = 'CONFLICT';
 export const CONVERSATION_URL = 'CONVERSATION_URL';
 
-export const ERROR = 'ERROR';
+export const _ERROR_ = 'ERROR';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: Same

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 13]
```diff
@@ -65,7 +65,7 @@ export const WEBEX_DOT_COM = 'webex.com';
 export const _ANSWER_ = 'ANSWER';
 export const _ACTIVE_ = 'ACTIVE';
 
-export const BUSY = 'BUSY';
+export const _BUSY_ = 'BUSY';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: Same

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js [hunk 14]
```diff
@@ -5,7 +5,7 @@ import {
   SIP_URI,
   PERSONAL_ROOM,
   MEETING_ID,
-  CONVERSATION_URL,
+  _CONVERSATION_URL_,
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/util.js]: Is there another `CONVERSATION_URL` that this one needed changed.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 15]
```diff
@@ -573,3 +569,9 @@ export const SDP = {
   CARRIAGE_RETURN: '\r\n',
   BAD_MEDIA_PORTS: [9, 0]
 };
+
+export const HTTP_VERBS = {
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: 👍 This is a good example of the grouping I spoke of in other comments, eliminating the need for `_PUT_`, etc.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 16]
```diff
@@ -74,10 +74,10 @@ export const _CONVERSATION_URL_ = 'CONVERSATION_URL';
 
 export const _ERROR_ = 'ERROR';
 
-export const IDLE = 'IDLE';
-export const IN_LOBBY = 'IN_LOBBY';
-export const IN_MEETING = 'IN_MEETING';
-export const INCOMING = 'INCOMING';
+export const _IDLE_ = 'IDLE';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: Are these duplicated elsewhere?

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js [hunk 17]
```diff
@@ -1,5 +1,5 @@
 
-// import {_CALL_, MEETING, MEETING_CENTER, WEBEX_MEETING, UNKNOWN} from '../constants';
+// import {_CALL_, _MEETING_, _MEETING_CENTER_, WEBEX_MEETING, UNKNOWN} from '../constants';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js]: Remove all comments?

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 18]
```diff
@@ -98,11 +98,11 @@ export const _OFFER_ = 'OFFER';
 export const _OBSERVE_ = 'OBSERVE';
 export const _OTHER_ = 'OTHER';
 
-export const PERSONAL_ROOM = 'PERSONAL_ROOM';
+export const _PERSONAL_ROOM_ = 'PERSONAL_ROOM';
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: What is "ms", "os", "ps", etc.?

File: packages/node_modules/@webex/plugin-meetings/.eslintrc.json [hunk 19]
```diff
@@ -1,5 +1,6 @@
 {
   "rules": {
-    "no-underscore-dangle": "off"
+    "no-underscore-dangle": "off",
+    "prefer-arrow-callback": "off"
```
[the remark this answers is not in the document]
Comment (mattnorris) [reply packages/node_modules/@webex/plugin-meetings/.eslintrc.json]: Please remove this rule.

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattnorris) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/index.js]: Please remove this comment.

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattnorris) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/infoUtils.js]: Please remove these comments.

**Comment by mattnorris** [comment]
There are 2 things to complete before merging this: 
  - Remove the eslint rule. 
  - Remove the commented code. 

Regarding the underscored variables, what was the driver for this? I see some were done to distinguish one variable from another, e.g., `CALL` v. `_CALL_`, but some just seem to have them. I don't think this needs done in this PR, but I think after this is merged, cleaning up the constants is merited. I think it's better to group items within objects to distinguish them, e.g., `CALL` v. `LOCUS.CALL`.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js (hunk 18 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/constants.js]: fix based on letters at a time :P

**Comment by chrisadubois** [comment]
@mattnorris -- there are comments noting the delineating of underscored constants

**Comment by arun3528** [comment]
@mattnorris  moved the eslint rule and removed the comments. Going to FIx the automation in the next PR

**Pull request merged**